src/Integration/Binding/CSharpVBBindingOperation.cs

```diff
@@ -142,6 +142,39 @@ public void Commit()
         #region Helpers
 
         private void AddRuleset()
+        {
+            TrySetNonConditionalRuleSet();
+            SetConditionalRuleSets();
+        }
+
+        /// <summary>
+        /// Try to set a ruleset without specifying a configuration (e.g. Debug / Release).
+        /// No changes are made if the project already has a ruleset.
+        /// </summary>
+        private void TrySetNonConditionalRuleSet()
+        {
+            var hasUserSpecifiedRuleSet = propertyInformationMap.Keys.Any(x => !IsDefaultMicrosoftRuleSet(x.Value as string));
+
+            if (hasUserSpecifiedRuleSet)
+            {
+                // We could've done this for all projects that have configurations that point to the same ruleset but we don't want to create noise for existing users.
+                return;
+            }
+
+            var ruleSetFullFilePath = propertyInformationMap.Values.FirstOrDefault()?.NewRuleSetFilePath;
```

Comment:
Fetching the ruleset value from the property map doesn't look right. If we get this far we know that all of the `CodeAnalysisRuleSet` properties (if any) refer to the default MS ruleset. So we know that we point to the solution-level ruleset i.e. `cSharpVBBindingConfig.RuleSet.Path` (which is what `propertyInformationMap.Values...NewRuleSetFilePath` should be returning anyway).